pyp2rpm/filters.py

```diff
@@ -71,11 +74,49 @@ def macroed_url(url):
     return url
 
 
+def rpm_version_410(version, use_macro=True):
+    if use_macro:
+        default = '%{pypi_version}'
+    else:
+        default = version
+    re_match = re.compile(
+        r"(\d+\.?\d*\.?\d*\.?\d*)\.?((?:a|b|rc|dev)\d*)").search(
+            version)
+    if not re_match:
+        return default
+    rpm_version, rpm_suffix = re_match.groups()
+    if rpm_suffix.startswith("dev"):
+        return '{}~~{}'.format(rpm_version, rpm_suffix)
+    else:
+        return '{}~{}'.format(rpm_version, rpm_suffix)
+
+
+def rpm_version(version, use_macro=True):
+    if use_macro:
+        default = '%{pypi_version}'
+    else:
+        default = version
+    re_match = re.compile(
+        r"(\d+\.?\d*\.?\d*\.?\d*)\.?((?:a|b|rc|post|dev)\d*)").search(
```

Comment:
Why does this regex contain `post` as choice, but the one above does not?
And should we actually support `post` and `dev` versions in pyp2rpm?

Reply:
I've added documentation to the filter functions.  rpm 4.10 - 4.14 support pre-release suffixes.  4.15 added support for a post-release suffix.